### PR TITLE
ref(react): Fix/refactor some Form components

### PR DESCRIPTION
**Pull request by billyvg**
* Change form fields to accept inline style props
* Fix HTML for `<MultipleCheckboxField>` to be more similar to existing form
* Fix form cancel button from submitting
* Fix FormField className

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/static/sentry/app/components/forms/formField.jsx [hunk 1]
```diff
@@ -99,28 +102,27 @@ export default class FormField extends React.Component {
...
-          {this.props.disabled &&
-            this.props.disabledReason &&
-            <span className="disabled-indicator tip" title={this.props.disabledReason}>
+          {disabled &&
+            disabledReason &&
+            <span className="disabled-indicator tip" title={disabledReason}>
               <span className="icon-question" />
             </span>}
-          {defined(this.props.help) && <p className="help-block">{this.props.help}</p>}
+          {defined(help) && <p className="help-block">{help}</p>}
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/components/forms/formField.jsx]: didn't know we had `defined`, nice

File: src/sentry/static/sentry/app/components/forms/multipleCheckboxField.jsx [hunk 2]
```diff
@@ -43,35 +45,55 @@ export default class MultipleCheckboxField extends FormField {
...
-            this.props.disabledReason &&
-            <span className="disabled-indicator tip" title={this.props.disabledReason}>
-              <span className="icon-question" />
-            </span>}
-        </label>
-        {this.props.help && <p className="help-block">{this.props.help}</p>}
-        {error && <p className="error">{error}</p>}
-        <div className="controls control-list">
-          {this.props.choices.map(([value, label]) => {
+      <div style={style} className={cx}>
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/components/forms/multipleCheckboxField.jsx]: why cx? is this a shorthand for classname? hadn't seen it

File: src/sentry/static/sentry/app/components/forms/multipleCheckboxField.jsx [hunk 3]
```diff
@@ -43,35 +45,55 @@ export default class MultipleCheckboxField extends FormField {
...
+        <div className={labelCx}>
+          <div className="controls">
+            <label className="control-label">
+              {label}
+              {shouldShowDisabled &&
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/components/forms/multipleCheckboxField.jsx]: nit, maybe name this shouldShowReason or shouldShowDisabledReason?

**Review by MaxBittker — DISMISSED** [review]
lgtm, had non-blocking nit

File: src/sentry/static/sentry/app/components/forms/multipleCheckboxField.jsx (hunk 2 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/forms/multipleCheckboxField.jsx]: just a short var name for class name, I saw it used when the `classnames` lib first came out and it just stuck w/ me. helps when you destructure `className` from `props`.

Comment (MaxBittker) [reply src/sentry/static/sentry/app/components/forms/multipleCheckboxField.jsx]: hm, i like it

**Review by MaxBittker — APPROVED** [review]
✅  nice fixes

**Pull request merged**